packages/odyssey-react-mui/src/Dialog.tsx

```diff
@@ -139,9 +139,17 @@ const Dialog = ({
   ].filter(Boolean);
 
   return (
-    <MuiDialog data-se={testId} open={isOpen} onClose={onClose}>
-      <DialogTitle translate={translate}>
-        {title}
+    <MuiDialog
+      data-se={testId}
+      open={isOpen}
+      onClose={onClose}
+      aria-labelledby="ods-dialog-title"
+    >
+      <DialogTitle
+        id="" // mui automatically set id based on aria-labelledby from MuiDialog, explicitly unset it to use id from span
+        translate={translate}
+      >
```

Comment:
Now I get what's going on here. I wanna make this less hacky.

Instead of adding a `span` with an unchanging `id` (which will cause conflicts with other Dialogs on the page), we should add `ariaLabel` to `DialogTitle`.

`ariaLabel` takes precedence over the text content:

```suggestion
      <DialogTitle
        ariaLabel={title} // This removes the "close" button as part of the label.
        translate={translate}
      >
```

Reply:
tested with voiceOver, ariaLabel not work as expected